infra/blockchain/cfxFetcher.go

```diff
@@ -0,0 +1,91 @@
+package blockchain
+
+import (
+	sdk "github.com/Conflux-Chain/go-conflux-sdk"
+	"github.com/Conflux-Chain/go-conflux-sdk/bind"
+	"github.com/Conflux-Chain/go-conflux-sdk/types"
+	"github.com/Conflux-Chain/go-conflux-sdk/types/cfxaddress"
+	"github.com/ethereum/go-ethereum/common"
+	evmTypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/sirupsen/logrus"
+	"github.com/tree-graph/bridge-service/infra/contracts/tokens"
+	"github.com/tree-graph/bridge-service/infra/contracts/vault"
+	"time"
+)
+
+type CfxFetcher struct {
+	Client     sdk.Client
+	filter     *tokens.TokenVaultFilterer
+	filterOpts *bind.FilterOpts
+	vaultAddr  string
+}
+
+func GetCfxClient() *sdk.Client {
+	return nil
+}
+
+func NewCfxFetcher(chain int64, vaultAddr string, client *sdk.Client) (*CfxFetcher, error) {
+	filter, err := tokens.NewTokenVaultFilterer(cfxaddress.MustNewFromHex(vaultAddr, uint32(chain)), client)
+	if err != nil {
+		logrus.WithError(err).Error("NewTokenVaultFilterer by cfx fail")
```

Comment:
Better to just return the error and let the business caller decide whether to log this error. To return more detailed error message, maybe you can use `errors.WithMessage(err, "detailed error message")`

Reply:
Fixed.